frontend/src/metabase/lib/markdown/parseMarkdown.tsx

```diff
@@ -0,0 +1,16 @@
+import ReactDOMServer from "react-dom/server";
```

Comment:
This is not something we'd want to use on the client

Reply:
> This is not something we'd want to use on the client

Why not?

BTW `frontend/src/metabase/static-viz/index.js` already uses it.

> `:first-line` selector seems promising https://caniuse.com/css-first-line

How would we use it?
Keep in mind that:
- [not all CSS properties](https://developer.mozilla.org/en-US/docs/Web/CSS/::first-line#allowable_properties) can be used with it.
- markdown may contain non-text nodes, e.g. images (see unit tests for it) - if the first node is not a text node `:first-line` won't help

> Instead of doing all the parsing, could you rely on CSS instead? 

I'm open for suggestions but I see nothing wrong with the current approach.

Reply:
Static viz uses it on the server, not in browser

Reply:
As for the explanation why I think it's a problematic approach. `react-dom/server` is a heavy dependency, it relies on the nodejs runtime. It works because we use webpack which polyfills used native nodejs modules, but it seems to be an overkill for a cosmetic change. 

If this task is something that cannot be achieved with a simpler approach, let's start a discussion with product people and see if there are alternatives.

Reply:
@kamilmielnik tagging again just to make sure you get a notification - I think the new approach from the comment is good

Reply:
41e627ef35...d702b570d19da2a29f282dbc577d0bf361d4ad7a